<p>This script takes the sequences you used to represent your OTUs and assigns them to a real-world genus, species, or other taxonomic unit.<br/><br/><em>NOTE: This is typically a very long step, and can take several minutes to complete.</em></p>

<p><strong>Here are scripts to execute before this one</strong>
<ul>
<li><a onclick="displayHideables('pick_rep_set')">Pick representative OTU sequences</a> - The sequences you will assign real-world taxa to are generated and stored in a fasta file</li>
</ul>
</p>

<p><strong>Here are the main decisions you should make before running this script</strong>
<ol>
<li>Pick an input file (.fasta format, one sequence per OTU in your study).</li>
<li>Pick an option for the --assignment_method parameter.  The different options each have their own parameters.</li>
<ul>
	<li><a target="_BLANK" href="http://nar.oxfordjournals.org/content/early/2013/11/26/nar.gkt1244.full">rdp</a></li>
	<li><a target="_BLANK" href="http://www.nature.com/scitable/topicpage/basic-local-alignment-search-tool-blast-29096">blast</a></li>
	<li><a target="_BLANK" href="http://www.nature.com/ismej/journal/v6/n7/abs/ismej2011208a.html">rtax</a></li>
	<li><a target="_BLANK" href="http://aem.asm.org/content/75/23/7537.full">mothur</a></li>
	<li><a target="_BLANK" href="http://www.ncbi.nlm.nih.gov/pubmed/22134646">tax2tree</a> - designed for use with Greengenes</li>
	<li><a target="_BLANK" href="http://bioinformatics.oxfordjournals.org/content/26/19/2460.full">uclust</a></li>
</ul>
<li>Pick a set of reference files</li>
<li>Pick output options</li>
</ol>
</p>

<p><strong>Here are the scripts to run after this one</strong>
<ul>
<li><a onclick="">Make OTU table</a> - The file generated by this script is an optional parameter in Make OTU table.  Including it adds each OTUs taxa to the table as metadata.</li>
</ul>
</p>

