<p>This is the final step of QIIME's phylogeny analysis.  It creates a .tre phylogenetic tree file that you can import into another program for viewing/manipulation</p>

<p><strong>Here are scripts to execute before this one</strong>
<ul>
<li><a onclick="displayHideables('align_seqs');">Align sequences</a> - Input for this script is a modified fasta file.  It contains headers and sequences, but some bases are replaced with -'s if they are identical to the reference alignment.  The script align sequences creates such a fasta file</li>
<li><a onclick="displayHideables('filter_alignment');">Filter alignment</a> - Creating the tree will be much more efficient if you filter out uninformative base-pair positions</li>
</ul>
</p>

<p><strong>Here are the main decisions you should make before running this script</strong>
<ol>
<li>Specify an --input_fp (the modified fasta file created from sequence alignment)</li>
<li>Choose a --tree_method</li>
<ul>
	<li><a target="_BLANK" href="http://www.ncbi.nlm.nih.gov/pubmed/16982706">clearcut</a></li>
	<li><a target="_BLANK" href="http://www.ncbi.nlm.nih.gov/pubmed/17846036">clustalw</a></li>
	<li><a target="_BLANK" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2693737/">fasttree_v1</a></li>
	<li><a target="_BLANK" href="http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0009490">fasttree</a></li>
	<li><a target="_BLANK" href="http://bioinformatics.oxfordjournals.org/content/22/21/2688.short">raxml_v730</a></li>
	<li><a target="_BLANK" href="http://www.ncbi.nlm.nih.gov/pubmed/15034147">muscle</a></li>
</ul>
<li>Choose a --root_method.  Typically the tree_method_default will be sufficient, but in some circumstances, you may want to use midpoint instead</li>
<li>Pick output options.<br/><em>Note: If you don't specify a --log_fp, no log file will be created</em></li>
</ol>
</p>

<p><strong>Here are the scripts to run after this one</strong>
<ul>
<li>None - this is the end of QIIME's phylogeny analysis</li>
</ul>
</p>

<p><strong>Here are some of the common pitfalls using this script</strong>
<ul>
<li>If you don't specify a log --file_fp, no log file will be created</li>
</ul>
</p>
