<p>It is highly economical to sequence multiple samples on a single NGS run. This script splits reads from a single fasta file (a single run) into their respective samples. It also can perform quality filtering at the same time</p>

<p><strong>Here are scripts to execute before this one</strong>
<ul>
<li><a onclick="displayHideables('validate_mapping_file')">Validate mapping file</a> - This script relies on a mapping file to accurately tell which samples correspond to which sequence barcodes.  You may want to check the format of your file before running this script</li>
<li><a onclick="displayHideables('join_paired_ends')">Join paired ends</a> - This script only works on a single fasta file.  QIIME provides you a script to join paired end reads into a single file</li>
<li><a onclick="displayHideables('extract_barcodes')">Extract barcodes</a> - If your fastq file includes barcode information, this script will create the necessary barcode read file</li>
</ul>
</p>

<p><strong>Here are the main decisions you should make before running this script</strong>
<ol>
<li>Select input files<ul>
	<li>--sequence_read_fps: .fastq files</li>
	<li>--mapping_fp: a validated map file describing barcodes for samples</li>
</ul></li>
<li>Pick barcode options.  You must either specify --barcode_read_fps, or else set barcode type to 'not barcoded'</li>
<li>Pick filtering options, either quality based or sequence based</li>
<li>Pick output options</li>
</ol>
</p>

<p><strong>Here are the scripts to run after this one</strong>
<ul>
<li><a onclick="displayHideables('pick_otus')">Pick OTUs</a> - Once your multiplexed reads have been assigned to samples, you are ready to select OTUs</li>
</ul>
</p>
