<p>Map files provide meta data about your samples.  The purpose of this script is to verify that the map file you have created is correctly formatted, and that it will not cause QIIME any problems later on in the pipeline.  It's not that we don't trust you; it's that we don't trust QIIME</p>

<p><strong>Here are scripts to execute before this one</strong>
<ul>
<li>None - This is the first script in the QIIME pipeline.  Before you do run this script, though, you must create a map file.  Notes about the map file format are listed below</li>
</ul>
</p>

<p><strong>Here are the main decisions you should make before running this script</strong>
<ol>
<li>Pick your map file, --mapping_fp.  <a href="index.php?step=upload">If you haven't uploaded your map file yet, do so here</a></li>
<li>If your map file has any known exceptions, indicate them (--not_barcoded, --variable_len_barcodes, --disable_primer_check)</li>
<li>Pick output options</li>
</ol>
</p>

<p><strong>Here are the scripts to run after this one</strong>
<ul>
<li><a onclick="displayHideables('split_libraries')">De-multiplex libraries</a> - If your sequence data is multiplexed (multiple samples in one sequencing lane to save money; samples differentiated by a unique sequence barcode), you must explicitly assign each read to a sample by using Split libraries</li>
<li><a onclick="displayHideables('join_paired_ends')">Join paired ends</a> - If your sequence has paired ends (two fasta files with corresponding reads; this is likely if you used Illumina), you will need to join them using Join paired ends before you can continue the QIIME pipeline</li>
<li><a onclick="displayHideables('pick_otus')">Pick OTUs</a> - If your sequences are neither multiplexed, nor paired-ended, you can just straight to this step</li>
</ul>
</p>

<p><strong>Here are notes about the map file format</strong><br/>
<a target=\"_blank\" href=\"http://www.qiime.org/documentation/file_formats.html#metadata-mapping-files\">Full documentation can be found here</a>. Each row is a sample (except the first, which is a header), and each column (separated by tabs) is an attribute. Valid attributes include ids, barcode sequences, primer sequences, reverse primer sequences, descriptions, case-control status, potential sources of bias, etc.</p>

<p>This script checks for the following problems in your mapping file:
<ul>
	<li>Required headers (SampleID, BarcodeSqeuence, LinkerPrimerSequence,
	and Description) are not all present.  The first three must be in this order.
	Description must be the last column</li>
	<li>Duplicate IDs or barcode/primer pairs exist </li>
	<li>Sequences use non-IUPAC DNA characters</li>
	<li>Barcode sequences are different lengths</li>
	<li>Headers, data fields, or SampleIDs contain invalid characters:
	<ul>
		<li>Headers: alphanumerics underscores</li>
		<li>Data fields: alphanumerics spaces + - % . / : , ;</li>
		<li>SampleIDs: alphanumerics periods</li>
	</ul>
</ul></p>

<p><strong>Here are some of the common pitfalls using this script</strong>
<ul>
<li>If your mapping file contains illegal characters, this script will create a corrected mapping file.
	Be sure to use that corrected version in future scripts (not the original one, which this script will not change)</li>
</ul>
</p>
