<p>After aligning sequences for phylogenetic analysis, it is good to filter out uninformative bases. This script does just that</p>

<p><strong>Here are scripts to execute before this one</strong>
<ul>
<li><a onclick="displayHideables('align_seqs')">Align sequences</a> - This script requires an alignment, one that is in the format created by align sequences</li>
</ul>
</p>

<p><strong>Here are the main decisions you should make before running this script</strong>
<ol>
<li>Choose your input file, a fasta file modified for sequence alignment</li>
<li>Decide if you want to use a lanemask (a precompiled list of positions to keep or to remove). If you don't want to use one, you must use --suppress_lane_mask_filter</li>
<li>Decide on filtering criteria, --entropy_threshold, --allowed_gap_frac, --remove_outliers/--threshold. </li>
<li>Pick output options</li>
</ol>
</p>

<p><strong>Here are the scripts to run after this one</strong>
<ul>
<li><a onclick="displayHideables('make_phylogeny')">Make phylogeny</a> - This script is just a preliminary step to actually creating a phylogenetic tree</li>
</ul>
</p>
